api/src/services/export-services/export-utils.ts

```diff
@@ -62,12 +74,88 @@ export function getQueryParams(query: SQLStatement | Knex.QueryBuilder): { text:
  * @export
  * @return {*}
  */
-export function getJsonStringifyTransformStream(): Transform {
+export function getStreamCsvTransformStream(header: string, collectionCategories?: string[]): Transform {
+  let headerStreamed = false;
   const transformStream = new Transform({
     objectMode: true, // Expects objects
     transform(chunk, _encoding, callback) {
-      // Stringify the chunk and push it to the next stream
-      callback(null, JSON.stringify(chunk));
+      if (header && !headerStreamed) {
+        // This block is executed only once
+        // Push the headers into stream
+        this.push([header, ...(collectionCategories ?? [])].join(',') + '\r\n');
+        headerStreamed = true;
+      }
+      // process chunk and push it to the next stream
+      let addLine = '\r\n';
+      if (!chunk) {
+        addLine = '';
+      }
+      callback(null, chunk + addLine);
+    }
+  });
+
+  return transformStream;
+}
+
+/**
+ * Get a query data record transform stream, that expects objects and outputs csv.
+ *
+ * Note: The incoming data stream must yield objects, or this will throw an error.
+ *
+ * @export
+ * @returns {Transform}
+ */
+export function getCsvTransformStream(
```

Comment:
Parts of this function are generic, and could be re-used. But other parts, like the measurements part, seem to be specific to certain exports.

I wonder if this function should be generated by each of the individual export strategies. And then it could cater to exactly what data is needed by each one. Some of the generic parts could be util functions, and others could be inline for each export, as needed.

You could still return it, and it could still be called in the root export function. But then the root export function wouldn't need to know about measurements, or any other data specific to one export.

(Not sure how drastic of a change this would be, so this is possibly more of an enhancement/TODO for the next team, if you think its valid).